dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetFactoryIntegrationTest.java

```diff
@@ -0,0 +1,214 @@
+
+        assertEquals(2, files.size());
+
+        final List<String> filesInodes = files.stream().map((file) -> file.getInode()).collect(Collectors.toList());
+        assertTrue(filesInodes.contains(fileAsset1.getInode()));
+        assertTrue(filesInodes.contains(fileAsset2.getInode()));
+    }
+
+    /**
+     * Method to test: {@link FileAssetFactory#findFileAssetsByFolderInDB(Folder, User, boolean)}
+     * When: Create one Folder without files
+     * Should: return a empty list
+     *
+     * @throws IOException
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void shouldReturnEmptyList() throws DotDataException, DotSecurityException {
+        final Role backEndUserRole = APILocator.getRoleAPI().loadBackEndUserRole();
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role, backEndUserRole).nextPersisted();
+
+        final Host host = new SiteDataGen().nextPersisted();
+        final Folder folder = new FolderDataGen().site(host).nextPersisted();
+
+        this.addPermission(role, folder);
+        final List<Contentlet> files = fileAssetFactory.findFileAssetsByFolderInDB(folder, user, false);
+
+        assertTrue(files.isEmpty());
+    }
+
+    /**
+     * Method to test: {@link FileAssetFactory#findFileAssetsByFolderInDB(Folder, User, boolean)}
+     * When: Create a Folder and not add permission
+     * Should: throw a {@link DotSecurityException}
+     *
+     * @throws IOException
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void shouldThorDotSecurityException() throws DotDataException, DotSecurityException {
+        final Role backEndUserRole = APILocator.getRoleAPI().loadBackEndUserRole();
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role, backEndUserRole).nextPersisted();
+
+        final Host host = new SiteDataGen().nextPersisted();
+        final Folder folder = new FolderDataGen().site(host).nextPersisted();
+
+        fileAssetFactory.findFileAssetsByFolderInDB(folder, user, false);
+    }
+
+    private Contentlet createFileAsset(
+            final Folder folder,
+            final String fileName,
+            final String suffix)
+            throws IOException, DotSecurityException, DotDataException {
+        final java.io.File file = java.io.File.createTempFile(fileName, suffix);
+        FileUtil.write(file, "helloworld");
+
+        return  new FileAssetDataGen(folder, file).nextPersisted();
+    }
+
+    private void addPermission(
+            final Role role,
+            final Permissionable... permissionables)
+            throws DotDataException, DotSecurityException {
+
+        final User systemUser = APILocator.systemUser();
+
+        for (Permissionable permissionable : permissionables) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'permissionable' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280157)